paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/SafeOreBreak.java

```diff
@@ -0,0 +1,199 @@
+import com.programmerdan.minecraft.simpleadminhacks.framework.autoload.AutoLoad;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.WeakHashMap;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+import org.bukkit.GameMode;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.plugin.java.JavaPlugin;
+import vg.civcraft.mc.civmodcore.inventory.items.MaterialUtils;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSetting;
+import vg.civcraft.mc.civmodcore.players.settings.PlayerSettingAPI;
+import vg.civcraft.mc.civmodcore.players.settings.gui.MenuSection;
+
+public final class SafeOreBreak extends BasicHack {
+
+	@AutoLoad
+	private List<String> ores;
+
+	private final List<BooleanOreSetting> oreSettings = new ArrayList<>();
+	private final List<List<Material>> breakOres = new ArrayList<>();
+
+	private final Set<Player> receivedWarning = Collections.newSetFromMap(new WeakHashMap<>());
+
+	public SafeOreBreak(final SimpleAdminHacks plugin, final BasicHackConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public void onEnable() {
+		super.onEnable();
+
+		MenuSection mainMenu = plugin.getSettingManager().getMainMenu();
+
+		OUTER:
+		for (String ore : ores) {
+			String[] parts = ore.split(",");
+			Material[] materialParts = new Material[parts.length];
+
+			for (int i = 0; i < parts.length; i++) {
+				String part = parts[i];
+				materialParts[i] = MaterialUtils.getMaterial(part);
+				if (materialParts[i] == null) {
+					logger.warning("Invalid material '"  + parts[0] + "'. Skipping.");
+					continue OUTER;
+				}
+			}
+
+			breakOres.add(Arrays.asList(materialParts));
+			String materialName = PlainTextComponentSerializer.plainText() .serialize(Component.translatable(materialParts[0]));
+			BooleanOreSetting setting = new BooleanOreSetting(plugin, true,
+					"Safe " + materialName + " break",
+					"safeOreBreak_" + materialParts[0].getKey().getKey(),
+					"Prevents you from breaking " + materialName
+							+ " without a silk touch pickaxe.",
+					materialParts[0]);
+			oreSettings.add(setting);
+			PlayerSettingAPI.registerSetting(setting, mainMenu);
+		}
+	}
+
+	@Override
+	public void onDisable() {
+		super.onDisable();
+	}
+
+	@EventHandler(ignoreCancelled = true)
+	public void onOreBreak(BlockBreakEvent event) {
+		if (event.getPlayer().getGameMode() != GameMode.SURVIVAL) {
+			return;
+		}
+
+		ItemStack item = event.getPlayer().getInventory().getItemInMainHand();
```

Comment:
Question, is it always true that blocks will be broken by the item in the main hand?

Reply:
Yes